src/main/java/org/candlepin/subscriptions/files/YamlPropertySourceFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.files;
+
+import java.io.IOException;
+import java.util.Properties;
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.support.EncodedResource;
+import org.springframework.core.io.support.PropertySourceFactory;
+
+public class YamlPropertySourceFactory implements PropertySourceFactory {
```

Comment:
Thanks for giving this a whirl. Because in order to load this way means we need to add an additional key (`yaml`), and this is confusing, instead of creating this class, and `YamlTagProfile`, I'd rather simplify to just a factory method (e.g.

```java
  @Bean
  public TagProfile tagProfile() throws FileNotFoundException {
    Yaml parser = new Yaml(new Constructor(TagProfile.class));
    return parser.load(new FileInputStream(ResourceUtils.getFile("classpath:tag_profile.yaml")));
  }
```

Reply:
:+1: